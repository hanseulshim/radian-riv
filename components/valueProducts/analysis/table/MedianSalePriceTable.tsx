import React, { useState, useCallback } from 'react'
import Table from 'components/common/Table'
import { formatPrice } from 'utils'

interface Props {
  tableData: any[]
  month: string
  label: string
}

export default function MedianSalePriceTable({
  tableData,
  month,
  label
}: Props) {
  const [data, setData] = useState([])
  const [columns, setColumns] = useState([])

  const fetchData = useCallback(async () => {
    const cols = [
      {
        Header: label,
        columns: [
          {
            Header: 'Contract Date',
            accessor: 'contractDate',
            align: 'right',
            width: 130
          },
          {
            Header: 'Number Pendings',
            accessor: 'numberPendings',
            align: 'right',
            width: 85
          },
          {
            Header: 'Number of Sales',
            accessor: 'numberSales',
            align: 'right',
            width: 85
          },
          {
            Header: 'Median Value',
            accessor: row => {
              return formatPrice(row.medianValue)
            },
            align: 'right',
            width: 100
          }
        ]
      }
    ]
    if (month !== 'one') {
      cols[0].columns.splice(1, 1)
    }
    setColumns(cols)
    setData(tableData)
  }, [tableData])

  return (
    <div className={`table-wrapper ${month}`}>
      <Table columns={columns} data={data} fetchData={fetchData} />
    </div>
  )
}
