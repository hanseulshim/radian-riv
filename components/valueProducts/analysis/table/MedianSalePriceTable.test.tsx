import * as React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'

import MedianSalePriceTable from './MedianSalePriceTable'

describe('MedianSalePriceTable', () => {
  let component
  beforeAll(() => {
    const props = {
      tableData: [],
      month: 'two',
      label: 'two'
    }
    component = shallow(<MedianSalePriceTable {...props} />)
  })
  it('get table container', () => {
    expect(component.find('.table-wrapper')).toHaveLength(1)
    expect(component.find('.two')).toHaveLength(1)
  })
  it('should match snapshot', () => {
    expect(toJson(component)).toMatchSnapshot()
  })
})
