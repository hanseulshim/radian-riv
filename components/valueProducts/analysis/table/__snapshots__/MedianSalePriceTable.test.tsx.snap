// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MedianSalePriceTable should match snapshot 1`] = `
<div
  className="table-wrapper two"
>
  <Table
    columns={Array []}
    data={Array []}
    fetchData={[Function]}
  />
</div>
`;
